src/builder.rs

```diff
@@ -12,9 +12,32 @@ use std::path::{Path, PathBuf};
 use std::process::Stdio;
 use std::str;
 
+use dirs::state_dir;
+use once_cell::sync::OnceCell;
 use tempfile::{Builder, TempDir};
 use tokio::process;
 
+/// The returned `&'static Path` can be coreced to any lifetime.
+fn get_default_control_dir<'a>() -> Result<&'a Path, Error> {
+    static DEFAULT_CONTROL_DIR: OnceCell<Option<Box<Path>>> = OnceCell::new();
+
+    DEFAULT_CONTROL_DIR
+        .get_or_try_init(|| {
+            if let Some(state_dir) = state_dir() {
+                fs::create_dir_all(&state_dir).map_err(Error::Connect)?;
+
+                Ok(Some(state_dir.into_boxed_path()))
+            } else {
+                Ok(None)
+            }
+        })
+        .map(|default_control_dir| {
+            default_control_dir
+                .as_deref()
+                .unwrap_or_else(|| Path::new("/tmp"))
```

Comment:
I think we should _never_ fall back to `/tmp`. _That_ is a security problem waiting to happen. It should either fall back to something like `./`, or we should return an error.

Reply:
I have set the fallback path to `./`.

Since we uses `./` for control directory before this PR, I believe we should also use it as a fallback.